Assets/Scripts/UI/PickupPup/Core/PPUIController.cs

```diff
@@ -33,7 +35,11 @@ protected override void setReferences()
         {
             dogProfileObject.SetActive(false);
         }
-        if(popupPrompt)
+        if (dogProfileShelterObject != null)
```

Comment:
Don't need to check != null for children of UnityEngine.Object. You can simply say:
```
if(obj)
{
     ...
}
```

https://docs.unity3d.com/ScriptReference/Object-operator_Object.html